packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -481,73 +489,1754 @@ describe('NetworkController', () => {
       });
     });
 
-    [
-      NetworkType.mainnet,
-      NetworkType.goerli,
-      NetworkType.sepolia,
-      NetworkType.rpc,
-    ].forEach((networkType) => {
-      describe(`when the provider config in state contains a network type of "${networkType}"`, () => {
-        lookupNetworkTests({
-          expectedProviderConfig: buildProviderConfig({ type: networkType }),
-          initialState: {
-            providerConfig: buildProviderConfig({ type: networkType }),
-          },
-          operation: async (controller) => {
-            await controller.lookupNetwork();
-          },
-        });
-      });
-    });
+    [NetworkType.mainnet, NetworkType.goerli, NetworkType.sepolia].forEach(
+      (networkType) => {
+        describe(`when the provider config in state contains a network type of "${networkType}"`, () => {
+          describe('if the network was switched after the net_version request started but before it completed', () => {
```

Comment:
This diff is making things the changes difficult to see, but if you remove the "if the network was switched" `describe`s then it should be a little more palatable, and you should see that `rpc` was removed from this array and put in its own `describe`. I did this because the setup needs to be switched for the custom RPC tests: instead of creating two fake providers, where the first is an Infura provider and the second is a custom RPC provider, it's the other way around. I didn't think that littering the tests with `if` statements would make things more readable, but I could be convinced otherwise.